File: examples/sphere.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

from hilma import Mesh, sphere, savePly
import numpy

size = 1024

mesh = sphere(size/2, 10)
savePly("sphere.ply", mesh, False)

File: examples/sphere.cpp
#include <string>
#include <sstream>
#include <unistd.h>
#include <iostream>

#include "hilma/types/Mesh.h"
#include "hilma/ops/generate.h"
#include "hilma/io/ply.h"

int main(int argc, char **argv) {
    int size = 1024;

    hilma::Mesh mesh = hilma::sphere(size/2, 10.0, hilma::TRIANGLE_STRIP);
    hilma::savePly("sphere.ply", mesh, false);

    return 1;
}